src/Polly.Core/Timeout/TimeoutResilienceStrategy.cs

```diff
@@ -58,7 +58,7 @@ public TimeoutResilienceStrategy(TimeoutStrategyOptions options, TimeProvider ti
         _cancellationTokenSourcePool.Return(cancellationSource);
 
         // check the outcome
-        if (outcome.Exception is OperationCanceledException e && isCancellationRequested && !previousToken.IsCancellationRequested)
+        if (isCancellationRequested && outcome.Exception is OperationCanceledException e && !previousToken.IsCancellationRequested)
```

Comment:
Check the boolean first as it avoids the need for a type check in the majority of cases.